db/python/tables/participant.py

```diff
@@ -581,17 +581,13 @@ async def get_external_ids_by_participant(
             return {}
 
         _query = """
-        SELECT participant_id AS id, JSON_ARRAYAGG(external_id) AS external_ids
+        SELECT participant_id AS id, JSON_ARRAYAGG(external_id) AS external_ids_list
         FROM participant_external_id
         WHERE participant_id IN :pids
         GROUP BY participant_id
         """
         rows = await self.connection.fetch_all(_query, {'pids': participant_ids})
-        retval: dict[int, list[str]] = {}
-        for row in rows:
-            external_ids = parse_sql_dict(row['external_ids'])
-            retval[row['id']] = list(external_ids.values()) if external_ids else []
-        return retval
+        return {r['id']: json.load(r['external_ids_list']) for r in rows}
```

Comment:
```suggestion
        return {r['id']: json.loads(r['external_ids_list']) for r in rows}
```

Of course, the real problem is that `get_external_ids_by_participant()` doesn't have a test case, so the unit tests aren't telling you about this :smile:

Reply:
I should stop committing code on the train

Reply:
:rofl:

Reply:
Ooh funnily enough there is a test that covers it: `test.test_search.TestSample.test_search_mixed` which just failed, at least for my latest commit